server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java

```diff
@@ -0,0 +1,297 @@
+import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.filter.TimeFilter;
+import org.apache.iotdb.tsfile.read.filter.basic.Filter;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.utils.TsPrimitiveType;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SerializeUtilsTest {
+  @Test
+  public void serdesStringTest() {
+    String str = "abcd%+/123\n\t";
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serialize(str, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Assert.assertEquals(str, SerializeUtils.deserializeString(buffer));
+  }
+
+  @Test
+  public void serdesStringListTest() {
+    List<String> slist = Arrays.asList("abc", "123", "y87@", "9+&d\n");
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serializeStringList(slist, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Assert.assertEquals(slist, SerializeUtils.deserializeStringList(buffer));
+  }
+
+  @Test
+  public void serdesIntListTest() {
+    List<Integer> intlist = Arrays.asList(12, 34, 567, 8910);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serialize(intlist, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    List<Integer> anotherIntlist = new ArrayList<>();
+    SerializeUtils.deserialize(anotherIntlist, buffer);
+    Assert.assertEquals(intlist, anotherIntlist);
+  }
+
+  @Test
+  public void serdesIntSetTest() {
+    List<Integer> intlist = Arrays.asList(12, 34, 567, 8910);
+    Set<Integer> intSet = new TreeSet<>(intlist);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serialize(intSet, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    List<Integer> anotherIntlist = new ArrayList<>();
+    SerializeUtils.deserialize(anotherIntlist, buffer);
+    Assert.assertEquals(intlist, anotherIntlist);
+  }
+
+  @Test
+  public void serdesNodeTest() {
+    Node node = new Node("127.0.0.1", 6667, 1, 6535, 4678);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serialize(node, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Node anotherNode = new Node("127.0.0.1", 6667, 1, 6535, 4678);
+    SerializeUtils.deserialize(anotherNode, buffer);
+    Assert.assertEquals(node, anotherNode);
+  }
+
+  @Test
+  public void serdesINT32BatchDataTest() {
+    BatchData batchData = new BatchData(TSDataType.INT32);
+    int ivalue = 0;
```

Comment:
Actually, there are three types of BatchData, and two of them are subClass of BatchData. You should test them all.